src/store/slices/mainSlice.ts

```diff
@@ -94,14 +96,25 @@ export const addUserThank = createAsyncThunk<void, IUser, ThunkApiConfig>(
         return;
       }
 
-      await addUser(value);
+      await setFirestoreData(usersCollection, value);
       dispatch(setUser(value));
```

Comment:
I don't think it's necessary to do a dispatch here.
1. they will execute regardless of the answer you get.
2. I think it is better to perform this operation in extrareduce on fulfield.
3. I am also not sure that a separate reducer is needed for this operation.
4. Why store all user fields separately in the state, if you can store the whole received object and retrieve the necessary fields from it.